File: android/src/main/kotlin/com/adyen/checkout/flutter/components/ComponentPlatformEventHandler.kt
package com.adyen.checkout.flutter.components

import com.adyen.checkout.flutter.generated.CheckoutEventDTO
import com.adyen.checkout.flutter.generated.EventsStreamHandler
import com.adyen.checkout.flutter.generated.PigeonEventSink

internal class ComponentPlatformEventHandler : EventsStreamHandler() {
    var eventSink: PigeonEventSink<CheckoutEventDTO>? = null
        private set

    override fun onListen(arguments: Any?, sink: PigeonEventSink<CheckoutEventDTO>) {
        eventSink = sink
    }

    override fun onCancel(arguments: Any?) {
        eventSink = null
    }

    fun send(event: CheckoutEventDTO) {
        eventSink?.success(event)
    }
}

File: ios/adyen_checkout/Sources/adyen_checkout/components/ComponentPlatformEventHandler.swift
import Foundation

final class ComponentPlatformEventHandler: EventsStreamHandler {
    private var eventSink: PigeonEventSink<CheckoutEventDTO>?

    override func onListen(withArguments _: Any?, sink: PigeonEventSink<CheckoutEventDTO>) {
        eventSink = sink
    }

    override func onCancel(withArguments _: Any?) {
        eventSink = nil
    }

    func send(event: CheckoutEventDTO) {
        eventSink?.success(event)
    }
}
